jmclient/test/test_tx_creation.py

```diff
@@ -278,6 +279,51 @@ def test_spend_p2wsh(setup_tx_creation):
     txid = jm_single().bc_interface.pushtx(tx)
     assert txid
 
+def test_spend_freeze_script(setup_tx_creation):
+    ensure_bip65_activated()
+
+    wallet_service = make_wallets(1, [[3, 0, 0, 0, 0]], 3)[0]['wallet']
+    wallet_service.sync_wallet(fast=True)
+
+    mediantime = jm_single().bc_interface.rpc("getblockchaininfo", [])["mediantime"]
+
+    timeoffset_success_ticks_tests = [(2, False, 50), (2, True, 0), (-60*60*24*30, True, 0), (60*60*24*30, False, 0)]
+
+    for timeoffset, required_success, ticks in timeoffset_success_ticks_tests:
+        #generate keypair
+        priv = "aa"*32 + "01"
+        pub = unhexlify(bitcoin.privkey_to_pubkey(priv))
+        addr_locktime = mediantime + timeoffset
+        redeem_script = bitcoin.mk_freeze_script(pub, addr_locktime)
+        script_pub_key = bitcoin.redeem_script_to_p2wsh_script(redeem_script)
+        regtest_vbyte = 100
+        addr = bitcoin.script_to_address(script_pub_key, vbyte=regtest_vbyte)
+
+        #fund frozen funds address
+        amount = 100000000
+        funding_ins_full = wallet_service.select_utxos(0, amount)
+        funding_txid = make_sign_and_push(funding_ins_full, wallet_service, amount, output_addr=addr)
+        assert funding_txid
+
+        #spend frozen funds
+        frozen_in = funding_txid + ":0"
+        output_addr = wallet_service.get_internal_addr(1)
+        miner_fee = 5000
+        outs = [{'value': amount - miner_fee, 'address': output_addr}]
+        tx = bitcoin.mktx([frozen_in], outs, locktime=addr_locktime+1)
+        i = 0
+        sig = bitcoin.get_p2sh_signature(tx, i, redeem_script, priv, amount)
+
+        assert bitcoin.verify_tx_input(tx, i, script_pub_key, sig, pub,
+            scriptCode=redeem_script, amount=amount)
+        tx = bitcoin.apply_freeze_signature(tx, i, redeem_script, sig)
+        push_success = jm_single().bc_interface.pushtx(tx)
+
+        assert push_success == required_success
+
+        for i in range(ticks):
+            time.sleep(1)
```

Comment:
Wouldn't it be better to just tick forward all the blocks at once?

Reply:
Oh I see; you are using second-by-second timing in the locktimes to test the required behaviour. Hmm, is there a way we can avoid having a 50 second sleep wait though?

Reply:
The sleep is needed to make the header timestamp tick forward as well. If you tick forward all the blocks together then `mediantime` won't advance.

Reply:
Oh of course, doh. Still it is pretty unfortunate if we have an extra 50s wait here.